src/components/utils/include/utils/logger/logger_impl.h

```diff
@@ -30,43 +30,67 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_UTILS_INCLUDE_UTILS_LOG_MESSAGE_LOOP_THREAD_H_
-#define SRC_COMPONENTS_UTILS_INCLUDE_UTILS_LOG_MESSAGE_LOOP_THREAD_H_
+#ifndef SRC_COMPONENTS_UTILS_INCLUDE_UTILS_LOGGER_LOGGER_IMPL_H_
+#define SRC_COMPONENTS_UTILS_INCLUDE_UTILS_LOGGER_LOGGER_IMPL_H_
 
-#include <log4cxx/logger.h>
-#include <queue>
-#include <string>
+#include "utils/ilogger.h"
 
-#include "utils/macro.h"
+#include <functional>
+
+#include "utils/logger_status.h"
 #include "utils/threads/message_loop_thread.h"
 
 namespace logger {
 
-typedef struct {
-  log4cxx::LoggerPtr logger;
-  log4cxx::LevelPtr level;
-  std::string entry;
-  log4cxx_time_t timeStamp;
-  log4cxx::spi::LocationInfo location;
-  log4cxx::LogString threadName;
-} LogMessage;
-
 typedef std::queue<LogMessage> LogMessageQueue;
 
 typedef threads::MessageLoopThread<LogMessageQueue>
     LogMessageLoopThreadTemplate;
 
+template <typename T>
+using LoopTreadPtr = std::unique_ptr<T, std::function<void(T*)> >;
+
 class LogMessageLoopThread : public LogMessageLoopThreadTemplate,
                              public LogMessageLoopThreadTemplate::Handler {
  public:
-  LogMessageLoopThread();
-  ~LogMessageLoopThread();
-  void Handle(const LogMessage message) OVERRIDE;
+  LogMessageLoopThread(std::function<void(LogMessage)> handler)
+      : LogMessageLoopThreadTemplate("Logger", this), force_log_(handler) {}
+
+  void Push(const LogMessage& message) {
+    PostMessage(message);
+  }
+
+  void Handle(const LogMessage message) {
+    force_log_(message);
+  }
+
+  ~LogMessageLoopThread() {
+    // we'll have to drop messages
+    // while deleting logger thread
+    logger::logger_status = logger::DeletingLoggerThread;
+    LogMessageLoopThreadTemplate::Shutdown();
+  }
```

Comment:
It might be better to implement these methods in `logger_impl.cc` rather than here(in the header file).

Reply:
Fixed with https://github.com/smartdevicelink/sdl_core/pull/3472/commits/8b02276e58ae23d263122032f0fe292ab5614fbf